cli/cmd/status/status_test.go

```diff
@@ -28,39 +28,89 @@ import (
 	"k8s.io/client-go/kubernetes/fake"
 )
 
-// TestCheckConsulServers creates a fake stateful set and tests the checkConsulServers function.
-func TestCheckConsulServers(t *testing.T) {
-	c := getInitializedCommand(t, nil)
-	c.kubernetes = fake.NewSimpleClientset()
-
-	// First check that no stateful sets causes an error.
-	_, err := c.checkConsulServers("default")
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "no server stateful set found")
-
-	// Next create a stateful set with 3 desired replicas and 3 ready replicas.
-	var replicas int32 = 3
-
-	createStatefulSet("consul-server-test1", "default", replicas, replicas, c.kubernetes)
-
-	// Now we run the checkConsulServers() function and it should succeed.
-	s, err := c.checkConsulServers("default")
-	require.NoError(t, err)
-	require.Equal(t, "Consul servers healthy (3/3)", s)
-
-	// If you then create another stateful set it should error.
-	createStatefulSet("consul-server-test2", "default", replicas, replicas, c.kubernetes)
-	_, err = c.checkConsulServers("default")
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "found multiple server stateful sets")
-
-	// Clear out the client and now run a test where the stateful set isn't ready.
-	c.kubernetes = fake.NewSimpleClientset()
-	createStatefulSet("consul-server-test2", "default", replicas, replicas-1, c.kubernetes)
-
-	_, err = c.checkConsulServers("default")
-	require.Error(t, err)
-	require.Contains(t, err.Error(), fmt.Sprintf("%d/%d Consul servers unhealthy", 1, replicas))
+func TestCheckConsulAgents(t *testing.T) {
+	namespace := "default"
+
+	cases := map[string]struct {
+		clients int
+		servers int
+	}{
+		"No clients, no agents": {0, 0},
+		"3 clients, no agents":  {3, 0},
+		"3 clients, 3 agents":   {3, 3},
+		"No clients, 3 agents":  {3, 3},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			buf := new(bytes.Buffer)
+			c := setupCommand(buf)
+			c.kubernetes = fake.NewSimpleClientset()
+
+			// Before deployment, we shouldn't see the checks or errors.
+			err := c.checkConsulAgents(namespace)
+			require.NoError(t, err)
+
+			actual := buf.String()
+			require.NotContains(t, actual, "Consul Clients Healthy")
+			require.NotContains(t, actual, "Consul Servers Healthy")
+			buf.Reset()
+
+			if tc.clients != 0 {
```

Comment:
minor nit - might be something to think about for the next time.

this test would probably be easier to read if these were just part of the test cases.  I find it easier to read a state test that sets up and asserts one state, so the description and set up show it.  This rather shows the initial state, sets it up, asserts it, changes it, asserts it again.

So instead of these descriptions:
```
		"No clients, no agents",
		"3 clients, no agents",
		"3 clients, 3 agents"},
		"No clients, 3 agents",
```
you would have all the permutations of the states shown in this 2 state template:
```
		"No clients, no agents",
		"3 clients - 1 healthy, no agents",
		"3 clients - all healthy, no agents",
		"3 clients - 1 healthy, 3 agents - 1 healthy",
		"3 clients - all healthy, 3 agents - all healthy",
		"No clients, 3 agents - 1 healthy",
		"No clients, 3 agents - all healthy",

Reply:
I love that feedback. I implemented it and it cleans up a lot. Thank you!